components/web_discovery/browser/regex_util.h

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_WEB_DISCOVERY_BROWSER_REGEX_UTIL_H_
+#define BRAVE_COMPONENTS_WEB_DISCOVERY_BROWSER_REGEX_UTIL_H_
+
+#include <deque>
+#include <memory>
+#include <optional>
+#include <string>
+
+#include "base/containers/flat_map.h"
+#include "third_party/re2/src/re2/re2.h"
+
+namespace web_discovery {
+
+// Lazily creates and caches pre-compiled regexes, mainly used for
+// privacy risk assessment of page URLs/contents.
+class RegexUtil {
```

Comment:
IMO this one deserves a better name... I was having hard time to figure out why we are passing the util all around :D 
also it's meaning is not really about regex,  but rather about doing lookups for sensitive info? 
I'd also move the following methods away from this class 
```
  void RemovePunctuation(std::string& str);
  void TransformToAlphanumeric(std::string& str);
```

Reply:
also if this operates with only static global data, it's not shameful to make the whole thing a singletone and don't pass it everywhere

Reply:
removed the methods, but kept the name since it simply caches regex patterns